check_raid.pl

```diff
@@ -559,15 +558,41 @@ sub program_names {
 
 sub commands {
 	{
-		'status' => ['-|', '@CMD'],
+		'status' => [>&2', '@CMD'],
```

Comment:
Metastat will print to stdout if SVM is in use, however if it’s not configured it ‘there are no existing databases’ is sent to stderr.  I’ve not come across a Solaris 10 install that doesn’t have metastat installed by default, even when it’s not used, so capturing stderr enables allow the active sub to disable the plugin if SVM isn’t actually in use, rather than have it exit with an unknown status. I’m using the plugin on quite a large base with, amongst other things, mixtures of SVM and H/W raid, so having the plugin auto detect this status in this way means you don’t have to manually configure which plugins you want the script to run

I think ‘>&2’ is actually correct, it’s a special op used by the command function, cciss_vol_status does the same thing

Regards

Matt



On 16 Feb 2015, at 13:24, Elan Ruusamäe <notifications@github.com<mailto:notifications@github.com>> wrote:


In check_raid.pl<https://github.com/glensc/nagios-plugin-check_raid/pull/96#discussion_r24750127>:

> @@ -559,15 +558,41 @@ sub program_names {
>
>  sub commands {
>       {
> -             'status' => ['-|', '@CMD'],
> +             'status' => [>&2', '@CMD'],


this looks syntax error. what's your intention? does the metastat tools print data to stderr?

—
Reply to this email directly or view it on GitHub<https://github.com/glensc/nagios-plugin-check_raid/pull/96/files#r24750127>.


--------------------------------------------------------------------
Save Paper - Do you really need to print this e-mail?

Visit www.virginmedia.com for more information, and more fun.

This email and any attachments are or may be confidential and legally privileged
and are sent solely for the attention of the addressee(s). If you have received this
email in error, please delete it from your system: its use, disclosure or copying is
unauthorised. Statements and opinions expressed in this email may not represent
those of Virgin Media. Any representations or commitments in this email are
subject to contract. 

Registered office: Media House, Bartley Wood Business Park, Hook, Hampshire, RG27 9UP
Registered in England and Wales with number 2591237